controller/api/public/stat_summary_test.go

```diff
@@ -2,60 +2,49 @@ package public
 
 import (
 	"context"
+	"errors"
 	"reflect"
 	"testing"
-	"time"
 
-	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
 	tap "github.com/runconduit/conduit/controller/gen/controller/tap"
 	pb "github.com/runconduit/conduit/controller/gen/public"
 	"github.com/runconduit/conduit/pkg/k8s"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
 )
 
-type fakeProm struct{}
-
-func (f *fakeProm) Query(ctx context.Context, query string, ts time.Time) (model.Value, error) {
-	return model.Value(model.Vector{}), nil
-}
-func (f *fakeProm) QueryRange(ctx context.Context, query string, r promv1.Range) (model.Value, error) {
-	return model.Value(model.Vector{}), nil
-}
-func (f *fakeProm) LabelValues(ctx context.Context, label string) (model.LabelValues, error) {
-	return model.LabelValues{}, nil
-}
-func (f *fakeProm) Series(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) ([]model.LabelSet, error) {
-	return []model.LabelSet{}, nil
+type statSumExpected struct {
+	err     error
+	promRes model.Value
+	k8sRes  []runtime.Object
+	req     pb.StatSummaryRequest
+	res     pb.StatSummaryResponse
 }
 
-var (
-	fakeGrpcServer = newGrpcServer(
-		&mockTelemetry{},
-		tap.NewTapClient(nil),
-		fake.NewSimpleClientset(),
-		&fakeProm{},
-		"conduit",
-	)
-)
-
 func TestStatSummary(t *testing.T) {
 	t.Run("Successfully performs a query based on resource type", func(t *testing.T) {
-		expectations := map[pb.StatSummaryRequest]pb.StatSummaryResponse{
-			pb.StatSummaryRequest{
-				Selector: &pb.ResourceSelection{
-					Resource: &pb.Resource{
-						Type: k8s.KubernetesDeployments,
+		expectations := []statSumExpected{
+			statSumExpected{
+				err:     nil,
+				k8sRes:  []runtime.Object{},
```

Comment:
TIOLI -- this is the default value for k8sRes, so I think you could omit it. You could also omit `err: nil` in this struct. On a related note -- I don't see any instances where this field is set to anything other than an empty slice. Is the plan to start using it in tests in subsequent branches?

Reply:
I've removed `k8sRes` from the struct. The plan was indeed to start using it, but I'll omit it until we need it.

I'm going to leave `err: nil` in the code, though, as I prefer to see it called out explicitly, in the case someone adds another test case, they're more likely to be aware that `err` is available.